File: src/scripts/delete.py
from appwrite.client import Client
from appwrite.services.databases import Databases
import os
from dotenv import load_dotenv
load_dotenv()

client = Client()
client.set_endpoint(os.getenv('APPWRITE_ENDPOINT'))
client.set_project(os.getenv('APPWRITE_PROJECT'))
client.set_key(os.getenv('APPWRITE_API_KEY'))

DATABASE_ID = os.getenv('DATABASE_ID')
COLLECTION_ID = os.getenv('COLLECTION_ID')

def delete():
    database_id = DATABASE_ID
    collection_id = COLLECTION_ID

    databases = Databases(client)
    
    while True:
        # Fetch documents from the collection
        response = databases.list_documents(database_id, collection_id)
        documents = response['documents']
        
        if not documents:
            break  # Exit the loop if there are no more documents
        
        for document in documents:
            document_id = document['$id']
            try:
                # Delete each document by ID
                databases.delete_document(database_id, collection_id, document_id)
                print(f'Deleted document with ID: {document_id}')
            except Exception as e:
                print(f'Error deleting document with ID: {document_id} - {str(e)}')

if __name__ == '__main__':
    delete()

File: pkg/delete.go
package pkg

import (
	"fmt"
	"time"

	"github.com/appwrite/sdk-for-go/databases"
)

func DeleteDocs(appwrite_database *databases.Databases, database_id, collection_id string) {
	startTime := time.Now()
	fmt.Println("Running delete document API!")

	response, err := appwrite_database.ListDocuments(database_id, collection_id)
	if err != nil {
		fmt.Println("Error list document:", err)
		return
	}

	for i, document := range response.Documents {
		fmt.Printf("Document %d: ID = %s\n", i+1, document.Id)
		_, err := appwrite_database.DeleteDocument(database_id, collection_id, document.Id)
		if err != nil {
			fmt.Printf("Error deleting document %d: %s\n", i+1, err)
			continue
		}
		fmt.Printf("Document %d deleted successfully.\n", i+1)
	}

	elapsedTime := time.Since(startTime)
	fmt.Printf("All documents created successfully in %s.\n", elapsedTime)
}
